osv/packagist_version.py

```diff
@@ -0,0 +1,137 @@
+import re
+
+
+class PackagistVersion(object):
```

Comment:
Nit: no need for `object` here in Python3.

Incredible work! Are there any upstream tests we can port, from either Packagist or php itself to test version_compare and the packagist versioning?